modules/gapi/src/backends/ie/giebackend.cpp

```diff
@@ -805,6 +798,13 @@ void cv::gimpl::ie::GIEExecutable::run(cv::gimpl::GIslandExecutable::IInput  &in
         return;
     }
 
+    if (cv::util::holds_alternative<cv::gimpl::Exception>(in_msg))
+    {
+        // (4) If the Exception message is revieved, propagate it further.
+        out.post(std::move(cv::util::get<cv::gimpl::Exception>(in_msg)));
+        return;
+    }
```

Comment:
Should this be done at this level? As well as every other island's level?

Reply:
I mean, shouldn't this be handled uniformly for all possible islands? A level above.

Reply:
Couldn't agree more. Handle the input `exception` in a general way.